internal/controller/operatorconfigmap_controller.go

```diff
@@ -1075,3 +1090,81 @@ func parseTolerations(csiPluginTolerations string) ([]corev1.Toleration, error)
 	}
 	return tolerations, nil
 }
+
+func (c *OperatorConfigMapReconciler) removeNoobaa() error {
+	noobaa := &nbv1.NooBaa{}
+	noobaa.Name = "noobaa-remote"
+	noobaa.Namespace = c.OperatorNamespace
+
+	if err := c.get(noobaa); !meta.IsNoMatchError(err) && client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get remote noobaa: %v", err)
+	} else if noobaa.UID != "" && noobaa.GetDeletionTimestamp().IsZero() {
+		index := slices.IndexFunc(
+			noobaa.GetOwnerReferences(),
+			func(ref metav1.OwnerReference) bool {
+				return ref.Kind == "StorageClient"
+			},
+		)
+		if index != -1 {
+			noobaa.Spec.CleanupPolicy.AllowNoobaaDeletion = true
+			if err := c.update(noobaa); err != nil {
+				return fmt.Errorf("failed to update noobaa %v: %v", noobaa.Name, err)
+			}
+			if err := c.delete(noobaa); err != nil {
+				return fmt.Errorf("failed to delete remote noobaa: %v", err)
+			}
+		}
+	}
+	return nil
+}
+
+func (c *OperatorConfigMapReconciler) removeNoobaaOperator() error {
+
+	nb := &metav1.PartialObjectMetadataList{}
+	nb.SetGroupVersionKind(nbv1.SchemeGroupVersion.WithKind("NooBaa"))
+	if err := c.list(nb, client.Limit(1)); err != nil && !meta.IsNoMatchError(err) {
+		return fmt.Errorf("failed to list noobaa: %v", err)
+	}
+	if len(nb.Items) != 0 {
+		return nil
+	}
+
+	noobaaSubscription, err := c.getSubscriptionByPackageName("mcg-operator")
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	} else if noobaaSubscription == nil {
+		return nil
+	} else if noobaaSubscription.UID == "" {
+		return nil
+	}
```

Comment:
I see the two sub checks due to the getSub func, either `else if sub == nil || sub.UID == ""` or `else if kerrors.IsNotFound(err)` could replace this? former is more suitable i guess.

Reply:
done